### ISSUE TITLE
query-paths fails due to duplicate rows for symmetric metapaths

### ISSUE THREAD

**Issue by dhimmel** [greenelab/connectivity-search-backend#43]
I noticed a the frontend failed at https://search.het.io/?source=8224&target=32460 when I tried to select a metapath. Looks like a backend error. Specifically, https://search-api.het.io/v1/query-paths/?source=8224&target=32460&metapath=GpBPpG fails at:

https://github.com/greenelab/hetmech-backend/blob/df4cc76248c3e68cdcefee255b449f59fe9945d9/dj_hetmech_app/utils/paths.py#L29

**Comment by dhimmel** [comment]
Hmm seems to work locally at http://localhost:8000/v1/query-paths/?source=8224&target=32460&metapath=GpBPpG

**Comment by dhimmel** [comment]
From the debug HTML output:

```
Exception Type: MultipleObjectsReturned
Exception Value: get() returned more than one PathCount -- it returned 2!
```

We will have to see whether the database actually has two rows for this. My local runserver should be using the production database, so I am confused about the divergent behavior.

**Comment by dhimmel** [comment]
It looks like the issue is that some metapaths appear twice (in both orientations), when the source and target metanode are the same.

See this SQL output:

```
dj_hetmech=> select * from dj_hetmech_app_pathcount WHERE source_id = 8224 AND target_id = 32460;
    id     | path_count |       dwpc       |       p_value        | metapath_id | source_id | target_id |  dgp_id  
-----------+------------+------------------+----------------------+-------------+-----------+-----------+----------
  17923713 |          3 | 7.55260498685033 | 1.69844877470549e-07 | GpMFpG      |      8224 |     32460 |  5868899
  26183657 |         12 | 3.92799289176115 | 3.45864877871577e-44 | GeAeG       |      8224 |     32460 |  5289192
  42509492 |        146 |  6.3449725003302 | 8.40143420988342e-14 | GpBPpG      |      8224 |     32460 |  5813985
 144616987 |          3 |  9.2825420329641 |  7.6553238470034e-09 | GiGiG       |      8224 |     32460 |  5372217
 147418242 |         62 | 2.83959298331948 | 1.16397810205433e-08 | GeAeGiG     |      8224 |     32460 | 32179672
(5 rows)

dj_hetmech=> select * from dj_hetmech_app_pathcount WHERE source_id = 32460 AND target_id = 8224;
    id     | path_count |       dwpc       |       p_value        | metapath_id | source_id | target_id |  dgp_id  
-----------+------------+------------------+----------------------+-------------+-----------+-----------+----------
  17987718 |          3 | 7.55260498685033 |  1.6984487747053e-07 | GpMFpG      |     32460 |      8224 |  5869199
  26987844 |         12 | 3.92799289176115 | 3.45864877871577e-44 | GeAeG       |     32460 |      8224 |  5290527
  42603777 |        146 |  6.3449725003302 | 8.40143420988342e-14 | GpBPpG      |     32460 |      8224 |  5862663
 111874727 |        132 | 5.03510924596963 | 2.54731044029161e-05 | GpBPpGiG    |     32460 |      8224 | 34733684
 144697317 |          3 |  9.2825420329641 | 7.81456939352071e-09 | GiGiG       |     32460 |      8224 |  5373120
 147874998 |         46 | 3.41812537091104 | 5.80760497603504e-08 | GeAeGiG     |     32460 |      8224 | 32184199
(6 rows)
```

So it looks like the symmetric metapaths are duplicated (stored in both orientations), but not-nonsymetric. We see the duplication at

![image](https://user-images.githubusercontent.com/1117703/57046039-24570b80-6c3d-11e9-9c41-01f2e95d9964.png)

**Comment by dhimmel** [comment]
> My local runserver should be using the production database, so I am confused about the divergent behavior.

This issue was because locally I was not running the latest backend code. Once I updated the backend code, I got the error locally as well.

The error should no longer occur due to https://github.com/greenelab/hetmech-backend/pull/44. However, the database still has redundant rows for symmetric metapaths. https://github.com/hetio/hetmatpy/pull/13 would make it so future database imports will not have the duplicate rows.
